packages/flutter/test/widgets/two_dimensional_viewport_test.dart

```diff
@@ -2759,6 +2759,56 @@ void main() {
       expect(tester.getRect(find.byKey(const ValueKey<int>(1))),
           const Rect.fromLTWH(200.0, 200.0, 200.0, 200.0));
     }, variant: TargetPlatformVariant.all());
+
+    testWidgets('state is preserved after reordering',
+        (WidgetTester tester) async {
+      final TwoDimensionalChildBuilderDelegate delegate1 =
+          TwoDimensionalChildBuilderDelegate(
+              maxXIndex: 5,
+              maxYIndex: 5,
+              addAutomaticKeepAlives: false,
+              addRepaintBoundaries: false,
+              builder: (BuildContext context, ChildVicinity vicinity) {
+                ValueKey<int>? key;
+                if (vicinity == const ChildVicinity(xIndex: 1, yIndex: 1)) {
+                  key = const ValueKey<int>(1);
+                } else if (vicinity ==
+                    const ChildVicinity(xIndex: 1, yIndex: 2)) {
+                  key = const ValueKey<int>(2);
+                }
+                return Checkbox(key: key, value: false, onChanged: (_) {});
```

Comment:
See the styleguide about the anonymous parameter: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-anonymous-parameter-names